index.rst

```diff
@@ -5,77 +5,331 @@
+The missing piece of functionality is the capability of mixing a ``Watcher``-like behavior (event-processing) with ``ScriptQueue``/``OCPS``-like behavior.
+Furthermore, we are also interested in generating comprehensive reports from the results of these processes and alarms for particular conditions.
 
-As discussed in :sqr:`068` :cite:`SQR-068`, Sasquatch is based on InfluxDB OSS 2.x, a time series database. The new version of InfluxDB brings a number of important updates, amongst which two are relevant for stream processing; `Flux`_ and `InfluxDB task`_.
+The following are the key functionality we are looking for in the ``Catcher``.
 
-.. _Flux: https://docs.influxdata.com/flux/v0.x/
-.. _InfluxDB task: https://docs.influxdata.com/influxdb/v2.2/process-data/get-started/
+Commandable SAL Component
+-------------------------
 
+The ``Catcher`` needs to interact with the Rubin Observatory Control System, both by reading telemetry from the system and by commanding other CSCs (see :ref:`introduction-process-image-data`).
+We also expect to be able to interact with the ``Catcher`` in a similar way we interact with other components on the system, such that we can start, configure, stop and redeploy it the same way we do with other components.
 
-`Flux`_ is a new functional data scripting language that is designed to process data streams.
-It contains extensive libraries that allows users to express complex operations, compatible with what can be done with programming languanges.
-Furthermore, `InfluxDB task`_ provides a way to schedule `Flux`_ scripts.
+Furthermore, being a CSC we also expect it to be written using Python/salobj, which is a broadly used framework in the project. 
+This ensures we have enough knowledge in the project to maintain and support the ``Catcher`` development.
 
+One thing we must consider in this case is whether the ``Catcher`` CSC should be a single CSC that handles all conditions, or an indexed CSC were each instance handles a single case/rule.
+Using our experience with the ``Watcher`` it seems like having an indexed CSC would be preferable.
+Some reasons for this are:
 
-Each Catcher "rule" or "alarm" would be comprised of an `InfluxDB task`_ to execute a user defined `Flux`_ script periodically.
-The `Flux`_ script performs a set of actions/calculations based on EFD data, additional processes/tasks can be triggered depending on the results of these calculations and results can be stored in the EFD for future reference.
+*  A single CSC means there is a single event for each condition.
+   If the CSC is handling multiple conditions the information ends up getting "lost" in the data queue and it is not easily digestible by LOVE and other systems.
+   Whereas, if each instance of the ``Catcher`` handles a single condition/rule, events are well separated for each condition.
```

Comment:
Hmm, so I guess I never paid attention to this, but if `evt_thing1` is published by a CSC with 10 indexes (like the electrometer), the Catcher rules to specify the index (which is a parameter of the event topic).
That means the catcher will also publish an event (e.g. `evt_catcherAlert`) and whatever thing is acting on that also needs to know the index... 
I guess I didn't consider this much before, we'll need a good way to keep track of which index is which thing!

Reply:
Uhm.. not sure I fully understand what you are trying to say.. Maybe we should tag up in person?